libs/constrained-generators/src/Constrained/Examples/Map.hs

```diff
@@ -66,3 +66,11 @@ rangeSumSize = constrained $ \m ->
       , v ==. 1
       ]
   ]
+
+elemSpec :: Specification BaseFn (Int, Int, Map Int Int)
+elemSpec = constrained' $ \k v m ->
+  [ assert $ k `member_` dom_ m
+  , forAll' m $ \k' v' ->
+      whenTrue (k' ==. k) (v' ==. v)
+  , m `dependsOn` k
+  ]
```

Comment:
This is a clever technique to state that some key-value pair definitely appears in a map.

Reply:
I'm also working on implementing `lookup_` but that's currently stuck on an unrelated bug that's triggered by the propagate test. Once that's resolved I'll have that PR and you won't need to faff around with this technique.